src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -322,27 +308,40 @@ private static bool ValidateName(string name, bool targetingAttributes, ErrorSin
 
             var validName = TryValidateName(
                 name,
-                whitespaceError: Resources.FormatHtmlTargetElementAttribute_NameCannotBeNullOrWhitespace(targetName),
-                characterErrorBuilder: (invalidCharacter) =>
-                    Resources.FormatHtmlTargetElementAttribute_InvalidName(
-                        targetName.ToLower(),
-                        name,
-                        invalidCharacter),
+                targetName,
                 errorSink: errorSink);
 
             return validName;
         }
 
         private static bool TryValidateName(
             string name,
-            string whitespaceError,
-            Func<char, string> characterErrorBuilder,
+            string tagHelperName,
+            ErrorSink errorSink)
```

Comment:
No need for this overload. Have all callers pass the `restrictChildrenAttribute` argument.